runtime/src/account_saver.rs

```diff
@@ -40,36 +40,53 @@ fn max_number_of_accounts_to_collect(
         .sum()
 }
 
+// Due to the current geyser interface, we are forced to collect references to
+// `SanitizedTransaction` - even if that's not the type that we have.
+// Until that interface changes, this function takes in an additional
+// `txs_refs` parameter that collects references to `SanitizedTransaction`
+// if it's provided.
+// If geyser is not used, `txs_refs` should be `None`, since the work would
+// be useless.
 pub fn collect_accounts_to_store<'a, T: SVMMessage>(
     txs: &'a [T],
+    txs_refs: &'a Option<Vec<impl core::borrow::Borrow<SanitizedTransaction>>>,
```

Comment:
It may be owned or borrowed. If we have SanitizedTransaction we need only copy references. If we have a ResolvedTransactionView we need to convert to an owned SanitizedTransaction.

Of course this just needs to exist until geyser has a sane interface, but since that's blocked for now and we dont want to wait on it, we're stuck with something similar to this.

If we wanted to eat the cost for simplicity, we could also just clone our SanitizedTransaction

Reply:
'AsRef' unfortunately does not work, since it is not implemented for '&T'.